airflow/jobs/scheduler_job.py

```diff
@@ -684,11 +687,36 @@ def _process_dags(self, dagbag, dags, tis_out):
         :type dagbag: airflow.models.DagBag
         :param dags: the DAGs from the DagBag to process
         :type dags: List[airflow.models.DAG]
-        :param tis_out: A list to add generated TaskInstance objects
-        :type tis_out: list[TaskInstance]
-        :rtype: None
+        :rtype: list[TaskInstance]
+        :return: A list of generated TaskInstance objects
         """
         check_slas = conf.getboolean('core', 'CHECK_SLAS', fallback=True)
+
+        tis_out = []
+        dag_ids = [dag.dag_id for dag in dags]
```

Comment:
```suggestion
        dag_ids = {dag.dag_id for dag in dags}
```
This will turn it into a set :)

Reply:
The list is little faster. 
https://stackoverflow.com/questions/2831212/python-sets-vs-lists